### PR TITLE
Swithing to Python3

### PR DESCRIPTION
**Pull request by hef**
Switching to Python 3.
The package python-ldap was incompatible, so the ldap3 package is now used.

There is also some cleanup of old zoho code, with most of the remaining parts pulled into the new "signup" module

### CODE REVIEW

**Comment by hef** [comment]
Anyone testing this will probably have to wipe his or her existing vagrant instance.

<code>git clean -xfd</code> is probably not a bad idea either.

**Comment by justintconroy** [comment]
I created a superuser with manage.py createsuperuser, but it would appear that the superuser is not a staff member, so I can't test the member management features through the website normally. Did you intentionally change the way staff members are determined? How do I set that?

File: accounts/models.py [hunk 1]
```diff
@@ -65,21 +79,27 @@ def create_superuser(self, object_guid, password, email = None):
         user = self.create_user(object_guid, email=email, password=password)
         admins_dn = "CN={0},{1}".format("Domain Admins", settings.AD_BASEDN)
```
Comment (justintconroy) [inline accounts/models.py]: Maybe this line isn't working? The superusers aren't getting added to the "Domain Admins" group on the vagrant box. This was also true for an account I created in the django shell by calling create_superuser() directly. Where does admins_dn get used again?

**Comment by justintconroy** [comment]
Manually adding the account to the "Domain Admins" group using

    sudo samba-tool group addmembers "Domain Admins" <account_name>

seems to turn the account into a staff member correctly.

**Comment by justintconroy** [comment]
Are the various user fields on the member fields supposed to show a GUID instead of the username for each user? See:
userfield

and:

idcheckedby

Seems fairly user unfriendly, though I don't think this is actually specific to this pull request, as I noticed the first example when I was testing the member point list pull request. If it's not intentional, I'll just file a bug for it that we can fix later.

**Comment by justintconroy** [comment]
All `__unicode__()` methods defined in classes need to be replaced with `__str__()`, and the appropriate decorator should be used if we still want to be backward compatible with python 2. See [this](https://docs.djangoproject.com/en/1.7/topics/python3/#str-and-unicode-methods).

Not doing this breaks some of the name displays on the admin interface:

emailtemplate

**Comment by justintconroy** [comment]
Oh, hehe. The `__unicode__()` thing is probably why GUIDs are being shown instead of usernames everywhere. :-D

**Comment by justintconroy** [comment]
When attempting to send one of the templated emails, either from the member detail page or the admin interface, I get an exception. No similar problems were encountered when sending activation emails or the like. It says connection refused, but I can't really tell what it's trying to connect to:

Environment:


Request Method: POST
Request URL: http://127.0.0.1:8001/mm/person/1

Django Version: 1.7.1
Python Version: 3.4.1
Installed Applications:
('bootstrap3',
 'bootstrap3_datetime',
 'django.contrib.webdesign',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.sites',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'ckeditor',
 'djcelery',
 'reversion',
 'django_tables2',
 'ps1auth',
 'accounts',
 'rfid',
 'zoho_integration',
 'memberpoint',
 'member_management',
 'crm',
 'signup',
 'django.contrib.admin',
 'django.contrib.admindocs')
Installed Middleware:
('django.middleware.common.CommonMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware')


Traceback:
File "/home/vagrant/venv/lib/python3.4/site-packages/django/core/handlers/base.py" in get_response
  111.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/home/vagrant/venv/lib/python3.4/site-packages/django/contrib/auth/decorators.py" in _wrapped_view
  22.                 return view_func(request, *args, **kwargs)
File "/vagrant/member_management/views.py" in person_detail
  75.                     count = email_template.send(request.user, person)
File "/vagrant/member_management/models.py" in send
  244.             send_email.delay(self.pk, user.pk, target.pk, extra_context)
File "/home/vagrant/venv/lib/python3.4/site-packages/celery/app/task.py" in delay
  453.         return self.apply_async(args, kwargs)
File "/home/vagrant/venv/lib/python3.4/site-packages/celery/app/task.py" in apply_async
  555.             **dict(self._get_exec_options(), **options)
File "/home/vagrant/venv/lib/python3.4/site-packages/celery/app/base.py" in send_task
  353.                 reply_to=reply_to or self.oid, **options
File "/home/vagrant/venv/lib/python3.4/site-packages/celery/app/amqp.py" in publish_task
  305.             **kwargs
File "/home/vagrant/venv/lib/python3.4/site-packages/kombu/messaging.py" in publish
  168.                        routing_key, mandatory, immediate, exchange, declare)
File "/home/vagrant/venv/lib/python3.4/site-packages/kombu/connection.py" in _ensured
  457.                                            interval_max)
File "/home/vagrant/venv/lib/python3.4/site-packages/kombu/connection.py" in ensure_connection
  369.                         interval_start, interval_step, interval_max, callback)
File "/home/vagrant/venv/lib/python3.4/site-packages/kombu/utils/__init__.py" in retry_over_time
  243.             return fun(*args, **kwargs)
File "/home/vagrant/venv/lib/python3.4/site-packages/kombu/connection.py" in connect
  237.         return self.connection
File "/home/vagrant/venv/lib/python3.4/site-packages/kombu/connection.py" in connection
  741.                 self._connection = self._establish_connection()
File "/home/vagrant/venv/lib/python3.4/site-packages/kombu/connection.py" in _establish_connection
  696.         conn = self.transport.establish_connection()
File "/home/vagrant/venv/lib/python3.4/site-packages/kombu/transport/pyamqp.py" in establish_connection
  112.         conn = self.Connection(**opts)
File "/home/vagrant/venv/lib/python3.4/site-packages/amqp/connection.py" in __init__
  165.         self.transport = self.Transport(host, connect_timeout, ssl)
File "/home/vagrant/venv/lib/python3.4/site-packages/amqp/connection.py" in Transport
  186.         return create_transport(host, connect_timeout, ssl)
File "/home/vagrant/venv/lib/python3.4/site-packages/amqp/transport.py" in create_transport
  299.         return TCPTransport(host, connect_timeout)
File "/home/vagrant/venv/lib/python3.4/site-packages/amqp/transport.py" in __init__
  95.             raise socket.error(last_err)

Exception Type: OSError at /mm/person/1
Exception Value: [Errno 111] Connection refused

**Comment by justintconroy** [comment]
RFID seems to be working fine.

File: accounts/models.py (hunk 1 above)
Comment (hef) [inline accounts/models.py]: The bug was actually a few lines down, I need to add the user to the domain admins group, 
<code>c.modify(admins_dn, add_to_group_changelist)<code> is the correct line.  I had confused myself and was thinking I need to add the group to the user.

**Comment by hef** [comment]
@justintconroy  It looks like it's not communicating with the rabbitmq server.  I'll see If I can reproduce locally

**Comment by hef** [comment]
@justintconroy All of the issues you pointed out should be fixed.  Let me know if everything looks ok.

**Comment by justintconroy** [comment]
Perhaps this is kind of obvious from the interface when it happens, and maybe it doesn't really matter, but the same user account can check a member's ID twice. I don't think this will really be a big deal unless we have some other automated process that queries ID checks, but just giving a heads up.

**Comment by justintconroy** [comment]
I'm still getting the same 'Connection Refused' exception when I try to send email.

**Comment by justintconroy** [comment]
It doesn't work for me on the develop branch either, so it's not really unique to this pull request. It might just be a problem with my own setup.

**Comment by justintconroy** [comment]
So, as requested, I did the following in my vagrant instance:

    sudo yaourt -Sy --noconfirm --aur rabbitmq

That installed a bunch of things. Current status of rabbitmq:

    ● rabbitmq.service - RabbitMQ Messaging Server
       Loaded: loaded (/usr/lib/systemd/system/rabbitmq.service; disabled)
       Active: inactive (dead)

And, of course, I suppose because it's inactive, I still can't send emails (same Connection Refused error).

**Comment by hef** [comment]
That's an improvement, it's installed now.
Try turning it on:
    sudo systemctl start rabbitmq
    sudo systemctl enable rabbitmq

**Comment by hef** [comment]
PR #72 should fix any rabbitMQ issues you are having.  It literally removes rabbitmq.

**Pull request merged**